File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    set_lost = set(lost) - set(reserve)
    set_reserve = set(reserve) - set(lost)
    
    for res in set_reserve:
        if res-1 in set_lost:
            set_lost.remove(res-1)
        elif res+1 in set_lost:
            set_lost.remove(res+1)
    
    return n - len(set_lost)

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int cnt = 0;
        boolean[] avail = new boolean[n+1];
        Arrays.sort(lost);
        Set<Integer> set = new HashSet<>();
        for (int i=0;i<lost.length;i++) {
            set.add(lost[i]);
        }
        for (int i=0;i<reserve.length;i++) {
            avail[reserve[i]] = true;
        }
        
        for (int i=0;i<lost.length;i++) {
            int num = lost[i];
            
            if (avail[num]) {
                avail[num] = false;
                cnt++;
                continue;
            }
                             
            if (avail[num-1] && !set.contains(num-1)) {
                avail[num-1] = false;
                cnt++;
                continue;
            }
            if (num+1 <= n && avail[num+1] && !set.contains(num+1)) {
                avail[num+1] = false;
                cnt++;
            }
        }
        return n - lost.length + cnt;
    }
}